pulp_ansible/app/models.py

```diff
@@ -45,44 +43,67 @@ class Meta:
         unique_together = ("namespace", "name")
 
 
+class Tag(Model):
+    """A model representing a Tag."""
+
+    name = models.CharField(max_length=32, unique=True, editable=False)
+
+
 class CollectionVersion(Content):
     """
     A content type representing a Collection.
 
+    This model is primarily designed to adhere to the data format for Collection content. That spec
+    is here: https://docs.ansible.com/ansible/devel/dev_guide/collections_galaxy_meta.html
+
     Fields:
 
-        version (models.CharField): Version string.
-        metadata: Artifact metadata in JSON format.
-        contents: Contnet items (e.g. roles, modules) provided by
-            collection in JSON format.
-        quality_score: Total version quality score,
-            floating-point number in range [0, 5].
+        authors (psql_fields.ArrayField): A list of the CollectionVersion content's authors.
+        contents (psql_fields.JSONField): A JSON field with data about the contents.
+        description (models.TextField): A short summary description of the collection.
+        docs_blob (psql_fields.JSONField): A JSON field holding the various documentation blobs in
+            the collection.
+        documentation (models.URLField): The URL to any online docs.
+        homepage (models.URLField): The URL to the homepage of the collection/project.
+        issues (models.URLField): The URL to the collection issue tracker.
+        license (psql_fields.ArrayField): A list of licenses for content inside of a collection.
+        license_file (models.CharField): The path to the license file for the collection.
+        name (models.CharField): The name of the collection.
+        namespace (models.CharField): The namespace of the collection.
+        readme (models.TextField): The path to the Markdown (.md) readme file.
```

Comment:
Same for readme field. By the way:

Is it a path? If it's a path why it's TextField?
Is it a file contents? if so, please update docstring.

Reply:
The docstrings are copy/pastes from the site here so whatever is written there is what I think should be written here.  https://docs.ansible.com/ansible/devel/dev_guide/collections_galaxy_meta.html#structure

Reply:
It shouldn't be a `TextField` though that is a mistake. I'm going to fix that now. This should be a `CharField` since its the path.

Reply:
What's the point of storing data that is relevant only for importer?
What is actually relevant for program that calls importer is a content of this file.

Reply:
These docs read to users as if they describe the specification for a Collection. The idea is that those same users will expect those fields whenever they manage Collection content inside of pulp_ansible.

I agree some of these fields are silly, but if they need to be removed then shouldn't they be removed from the spec?

I can at the same time understand why this field is here, because what if a client pushing/pulling collection data wants to automated pull out the readme. It needs to know it's path to do that.

More comments/feedback is welcome.

Reply:
@bmbouter please see the explanation from me and Andrew above.